packages/editor/src/components/document-bar/index.js

```diff
@@ -48,40 +48,46 @@ const icons = {
 };
 
 export default function DocumentBar() {
-	const { isEditingTemplate, templateId, postType, postId } = useSelect(
-		( select ) => {
+	const { isEditingTemplate, templateId, postType, postId, goBack } =
+		useSelect( ( select ) => {
 			const {
 				getRenderingMode,
 				getCurrentTemplateId,
 				getCurrentPostId,
 				getCurrentPostType,
+				getEditorSettings,
 			} = select( editorStore );
 			const _templateId = getCurrentTemplateId();
+			const back = getEditorSettings().goBack;
 			return {
 				isEditingTemplate:
 					!! _templateId && getRenderingMode() === 'template-only',
 				templateId: _templateId,
 				postType: getCurrentPostType(),
 				postId: getCurrentPostId(),
+				goBack: typeof back === 'function' ? back : undefined,
 			};
-		},
-		[]
-	);
+		}, [] );
 	const { getEditorSettings } = useSelect( editorStore );
 	const { setRenderingMode } = useDispatch( editorStore );
 
+	const handleOnBack = () => {
+		if ( isEditingTemplate ) {
+			setRenderingMode( getEditorSettings().defaultRenderingMode );
+			return;
+		}
+		if ( goBack ) {
+			goBack();
+			return;
+		}
+		window.history.back();
+	};
+
 	return (
 		<BaseDocumentActions
```

Comment:
I think this component shouldn't be rendered if there's no "goBack" (and for now no "template mode" )